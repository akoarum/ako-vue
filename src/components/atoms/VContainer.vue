<script lang="ts">
import Vue, { CreateElement, VNode } from 'vue'

export default Vue.extend({
  props: {
    tag: { type: String, default: 'section' },
    color: { type: String }
  },
  computed: {
    styled(): { [key: string]: string } {
      const styles: { [key: string]: string } = {}

      if (this.color) {
        styles['--background'] = this.color
      }

      return styles
    }
  },
  render(createElement: CreateElement): VNode {
    return createElement(
      this.tag,
      { class: 'vContainer', style: this.styled },
      this.$slots.default
    )
  }
})
</script>

<style lang="scss" scoped>
.vContainer {
  --background: #fff;
  background: var(--background);

  @include media-min {
    padding: 24px 20px 28px;
  }

  @include media-max {
    padding: 20px 12px 23px;
  }
}
</style>
