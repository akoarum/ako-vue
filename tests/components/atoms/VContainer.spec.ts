import { shallowMount } from '@vue/test-utils'
import VContainer from '~/components/atoms/VContainer.vue'

describe('VContainer', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VContainer, {
      propsData: {},
      slots: {
        default: 'テキスト'
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styled: propsを基にスタイルを生成して返す', () => {
      expect(wrapper.vm.styled).toEqual({})

      wrapper.setProps({ color: '#f0f0f0' })
      expect(wrapper.vm.styled).toEqual({ '--background': '#f0f0f0' })
    })
  })
})
