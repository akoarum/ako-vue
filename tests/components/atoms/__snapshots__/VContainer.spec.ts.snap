// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VContainer initialized snapshot 1`] = `
<section
  class="vContainer"
>
  テキスト
</section>
`;
